org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/managers/GradleProjectImporterTest.java

```diff
@@ -58,4 +64,19 @@ public void testJavaImportExclusions() throws Exception {
 			javaImportExclusions.remove(GRADLE1_PATTERN);
 		}
 	}
+
+	@Test
+	public void testGradlePersistence() throws Exception {
+		importProjects("gradle/nested");
+		List<IProject> projects = ProjectUtils.getGradleProjects();
+		for (IProject project : projects) {
+			assertTrue(GradleProjectImporter.shouldSynchronize(project.getLocation().toFile()));
+		}
+		Job.getJobManager().join(CorePlugin.GRADLE_JOB_FAMILY, new NullProgressMonitor());
+		GradleBuildSupport.saveModels();
+		for (IProject project : projects) {
+			assertFalse(GradleProjectImporter.shouldSynchronize(project.getLocation().toFile()));
+		}
```

Comment:
you should also check that if the gradle files have been touched after saving the models then  they should synchronize